### DISCUSSION TITLE
Welcome to indexed.xyz Discussions!

### DISCUSSION THREAD

**Discussion by radiofreejohn** [indexed-xyz/docs#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas for things we can build together.
  * Share how you're using the indexed.xyz data.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by radiofreejohn** [comment]
Hey everyone! I'm John, the resident support person here at Goldsky. I'm here to help you get up and running with the Indexed.xyz dataset and various tools.

I like memes, turtles, and my cat Molly.

![molly](https://user-images.githubusercontent.com/299455/221362667-d2894a79-89d9-444e-823b-50181f7eff7a.png)

If you have a cat, please post cat pictures here as well, that's allowed, until it isn't.

**Reply by salieflewis** [reply]
Hi John!

**Reply by mseidlx** [reply]
Hey John, I think adding a simple Python step-by-step guide would be super helpful for many data analysts out there that aren't super familiar with S3 APIs :)

Also - I'm not sure if I fully understood the schema. E.g. is there an easy way to load all Ethereum transactions for the last 7d ?

-- unfortunately no cat pics :/

**Reply by hemaaanth** [reply]
Hey! We’ve added a python-based tutorial to our backlog! What do you think would be better, a notebook or a streamlit app?

**Reply by mseidlx** [reply]
A notebook sounds great!

**Reply by hemaaanth** [reply]
Hey! So sorry for the very late response here, lost this to-do in my backlog. Quick sample below.

Once you have the Goldsky CLI installed, getting a sample Base Goerli set of decoded logs:
```
goldsky indexed sync decoded-logs --network=base-goerli --contract-address=0xac6564f3718837caadd42eed742d75c12b90a052
```

Once this is downloaded and available as a bunch of parquet files in a folder on your machine, can set up a Python notebook with some of the base dependencies:
```
import pandas as pd
import os
import pyarrow.parquet as pq
```

And with this code (replace the directory structure as appropriate for your machine, you'll have the data available in a pandas dataframe! Can do the usual python magic from there.
```
directory = '<directory>' # Define your directory here

# Getting all the parquet files
file_list = [f for f in os.listdir(directory) if f.endswith('.parquet')]

# Load into a bunch of dataframes
dataframes = []
for file_name in file_list:
    file_path = os.path.join(directory, file_name)
    dataframe = pq.read_table(file_path).to_pandas()
    dataframes.append(dataframe)

# Concatenating into a single one
combined_dataframe = pd.concat(dataframes, ignore_index=True)
```

This is heavily not-optimized, and is best for single-contract exploration vs chain-level exploration, but should be good enough for small exploratory research. Let me know what you think and if it meets the needs / how you might want to improve it.
